data/plugins/treeview.lua

```diff
@@ -621,45 +587,103 @@ end
 
 local function treeitem() return view.hovered_item or view.selected_item end
 
-
-menu:register(function() return core.active_view:is(TreeView) and treeitem() end, {
-  { text = "Open in System", command = "treeview:open-in-system" },
-  ContextMenu.DIVIDER
-})
-
-menu:register(
-  function()
-    local item = treeitem()
-    return core.active_view:is(TreeView) and item and not is_project_folder(item)
-  end,
-  {
+function TreeView:on_context_menu()
+  return {
+    { text = "Open in System", command = "treeview:open-in-system" },
+    ContextMenu.DIVIDER,
     { text = "Rename", command = "treeview:rename" },
     { text = "Delete", command = "treeview:delete" },
-  }
-)
-
-menu:register(
-  function()
-    local item = treeitem()
-    return core.active_view:is(TreeView) and item and item.type == "dir"
-  end,
-  {
     { text = "New File", command = "treeview:new-file" },
     { text = "New Folder", command = "treeview:new-folder" },
-  }
-)
+    { text = "Remove directory", command = "treeview:remove-project-directory" },
+    { text = "Find in Directory", command = "treeview:search-in-directory" }
```

Comment:
Should this be guarded by `if projectsearch then`?

Reply:
Nah; if the command's predicate fails, or if the command doesn't exist, then this won't be added, so no need.